### ISSUE TITLE
Question/suggestion: Possible to set app-token-action reference as a variable

### ISSUE THREAD

**Issue by royveshovda** [rajbos/github-fork-updater#67]
We have places all our external actions in a separate organization for security reasons, and using you action to keep them up to date. Which is perfect for our use case. We have also updated to use the github app variant. However, you refer to an external action (https://github.com/peter-murray/workflow-application-token-action) which kind of breaks our model of controlling all our actions. This suggestion is to set this reference as an variable instead (at least as an option), and have this variable to point to an internal fork of the app-token repo.

Happy to provide a PR. Just wanted to verify you are OK with the idea first?

**Comment by rajbos** [comment]
Awesome to hear @royveshovda!

I get where the usage of that action can be an issue indeed, but I am not sure how we can make this variable for the end user. You can of course run the action from a local path, so maybe we can clone a repo based on a variable and then run it from there? That will break Dependabot version updates for the action, which does not help in that case. 

If you have a better suggestion, then I am all ears of course!

**Comment by royveshovda** [comment]
I will make an attempt on this, and send a PR for discussion. It might break in the ways you suggest, but lets have a look

**Comment by rajbos** [comment]
The hard part is of course that the runner parses the workflow files and downloads all actions before you get a change to do anything. I could support something where you inject either a custom script or a custom Docker container. That would create a lot of extra complexity for normal users.

At a customer we have fixed this during the sync of this repo to the internal list of allowed actions (which got mirror into their environment. Then we have a config file to overwrite certain lines of code, and the call to this action could be just that: over write it with the action you want to use. (Do note that keeping that action reference up to date is on you then). 

Closing this issue as I don't think either of us will work on this :-D.
